bcda/router_test.go

```diff
@@ -67,8 +72,30 @@ func (suite *RouterTestSuite) TestFileServerRoute() {
 	assert.Panics(suite.T(), func() {
 		FileServer(r, "/api/v1/swagger{}", http.Dir("./swaggerui"))
 	})
+}
+
+func (suite *RouterTestSuite) TestHTTPServerRedirect() {
+	// HTTP client that does not follow redirects
+	client := &http.Client{
+		Transport: &http.Transport{},
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	// Redirect GET http requests to https
+	res, err := client.Get(suite.httpServer.URL + "/")
+	assert.Nil(suite.T(), err, "redirect GET http to https")
+	assert.Equal(suite.T(), 301, res.StatusCode, "http to https redirect return correct status code")
+	assert.Contains(suite.T(), res.Header.Get("Location"), "https://", "location response header contains 'https://'")
 
```

Comment:
After reading [this doc](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security), I decided to move the HSTS header so that it is sent from the api/data servers when operating in `https` mode. From the doc:

> Note: The Strict-Transport-Security header is ignored by the browser when your site is accessed using HTTP; this is because an attacker may intercept HTTP connections and inject the header or remove it. When your site is accessed over HTTPS with no certificate errors, the browser knows your site is HTTPS capable and will honor the Strict-Transport-Security header.